lib/topological_inventory/orchestrator/worker.rb

```diff
@@ -1,124 +1,168 @@
+        object_manager.create_deployment_config(collector_deployment_name_for_source(source)) do |d|
+          d[:metadata][:labels]["topological-inventory/collector_digest"] = digest
+          d[:metadata][:labels]["topological-inventory/collector"] = "true"
+          d[:spec][:replicas] = 1
+          container = d[:spec][:template][:spec][:containers].first
+          container[:env] = collector_container_environment(source)
+          container[:image] = source["image"]
+        end
       end
 
-      def endpoints_for_source(source_id)
-        JSON.parse(RestClient.get(File.join(api_url, "api/v0.0/sources/#{source_id}/endpoints")))
+      def remove_openshift_objects_for_source(digest)
+        return unless digest
+        deployment = object_manager.get_deployment_configs("topological-inventory/collector_digest=#{digest}").detect { |i| i.metadata.labels["topological-inventory/collector"] == "true" }
+        return unless deployment
+        logger.info("Removing objects for deployment #{deployment.metadata.name}")
+        object_manager.delete_deployment_config(deployment.metadata.name)
+        object_manager.delete_secret("#{deployment.metadata.name}-secrets")
       end
 
-      def authentication_for_endpoint(endpoint_id)
-        conn = PG::Connection.new(pg_connection_args)
-        sql = <<~SQL
-          SELECT *
-          FROM authentications
-          WHERE
-            resource_type = 'Endpoint' AND
-            resource_id = $1
-        SQL
-        conn.exec_params(sql, [endpoint_id]).first.tap do |auth|
-          auth["password"] = MiqPassword.decrypt(auth["password"])
-        end
+      def collector_deployment_name_for_source(source)
+        "topological-inventory-collector-source-#{source["source_id"]}"
       end
 
-      def pg_connection_args
-        {
-          :host     => ENV["DATABASE_HOST"],
-          :port     => ENV["DATABASE_PORT"],
-          :dbname   => ENV["DATABASE_NAME"],
-          :user     => ENV["DATABASE_USER"],
-          :password => ENV["DATABASE_PASSWORD"]
-        }.freeze
+      def collector_deployment_secret_name_for_source(source)
+        "#{collector_deployment_name_for_source(source)}-secrets"
       end
 
-      def clear_caches
-        @sources = nil
-        @collector_deployments = nil
+      def collector_container_environment(source)
+        secret_name = "#{collector_deployment_name_for_source(source)}-secrets"
+        [
+          {:name => "AUTH_PASSWORD", :valueFrom => {:secretKeyRef => {:name => secret_name, :key => "password"}}},
+          {:name => "AUTH_USERNAME", :valueFrom => {:secretKeyRef => {:name => secret_name, :key => "username"}}},
+          {:name => "INGRESS_API", :value => "http://#{ENV["TOPOLOGICAL_INVENTORY_INGRESS_API_SERVICE_HOST"]}:#{ENV["TOPOLOGICAL_INVENTORY_INGRESS_API_SERVICE_PORT"]}"},
+          {:name => "SOURCE_HOST", :value => source["host"]},
+          {:name => "SOURCE_UID",  :value => source["source_uid"]},
```

Comment:
Can we make this match https://github.com/ManageIQ/topological_inventory-collector-openshift/pull/34